i6_models/parts/frontend/vgg.py

```diff
@@ -0,0 +1,382 @@
+            out_channels=model_cfg.conv4_channels,
+            kernel_size=model_cfg.conv_kernel_size,
+            padding=conv_padding,
+        )
+        self.activation = model_cfg.activation
+        if self.include_linear_layer:
+            self.linear = nn.Linear(
+                in_features=model_cfg.linear_input_dim,
+                out_features=model_cfg.linear_output_dim,
+                bias=True,
+            )
+
+    def forward(self, tensor: torch.Tensor, sequence_mask: torch.Tensor) -> Tuple[torch.Tensor, torch.Tensor]:
+        """
+        T might be reduced to T'' depending on the stride of the layers
+
+        :param tensor: input tensor of shape [B,T,F]
+        :param sequence_mask: masking tensor of shape [B,T], contains length information of the sequences
+        :return: torch.Tensor of shape [B,T',F']
+        """
+        tensor = tensor[:, None, :, :]  # [B,C=1,T,F]
+
+        tensor = self.conv1(tensor)
+        tensor = self.activation(tensor)
+        tensor = self.pool(tensor)  # [B,C,T',F']
+
+        tensor = self.conv2(tensor)  # [B,C,T",F"]
+        tensor = self.activation(tensor)
+
+        tensor = self.conv3(tensor)
+        tensor = self.activation(tensor)
+
+        tensor = self.conv4(tensor)
+        sequence_mask = _mask_pool(
+            sequence_mask, self.conv1.kernel_size[0], self.conv1.stride[0], self.conv1.padding[0]
+        )
+        sequence_mask = _mask_pool(sequence_mask, self.pool.kernel_size[0], self.pool.stride[0], self.pool.padding[0])
+        sequence_mask = _mask_pool(
+            sequence_mask, self.conv2.kernel_size[0], self.conv2.stride[0], self.conv2.padding[0]
+        )
+        sequence_mask = _mask_pool(
+            sequence_mask, self.conv3.kernel_size[0], self.conv3.stride[0], self.conv3.padding[0]
+        )
+        sequence_mask = _mask_pool(
+            sequence_mask, self.conv4.kernel_size[0], self.conv4.stride[0], self.conv4.padding[0]
+        )
+
+        tensor = torch.transpose(tensor, 1, 2)  # transpose to [B,T",C,F]
+        tensor = torch.flatten(tensor, start_dim=2, end_dim=-1)  # [B,T",C*F]
+
+        if self.include_linear_layer:
+            tensor = self.linear(tensor)
+
+        return tensor, sequence_mask
+
+
+def _get_padding(input_size: Union[int, Tuple[int, ...]]) -> Union[int, Tuple[int, ...]]:
+    """
+    get padding in order to not reduce the time dimension
+
+    :param input_size:
+    :return:
+    """
+    if isinstance(input_size, int):
+        return (input_size - 1) // 2
+    elif isinstance(input_size, tuple):
+        return tuple((s - 1) // 2 for s in input_size)
+    else:
+        raise TypeError(f"unexpected size type {type(input_size)}")
+
+
+def _mask_pool(seq_mask, kernel_size, stride, padding):
+    """
+    :param seq_mask: [B,T]
+    :param kernel_size:
+    :param stride:
+    :param padding:
+    :return: [B,T'] using maxpool
+    """
+    seq_mask = torch.unsqueeze(seq_mask, 1)  # [B,1,T]
```

Comment:
As mentioned in another commend (which you already clicked as resolved, although it is not resolved?), you miss a cast to int here, and then later a cast back to bool.
```suggestion
    seq_mask = torch.unsqueeze(seq_mask, 1)  # [B,1,T]
    seq_mask = seq_mask.int()  # max_pool needs integer
```